includes/class-wc-payments.php

```diff
@@ -945,6 +951,7 @@ private static function get_wc_payments_http() {
 
 		if ( ! $http_class instanceof WC_Payments_Http_Interface ) {
 			$http_class = new WC_Payments_Http( new Automattic\Jetpack\Connection\Manager( 'woocommerce-payments' ) );
+			$http_class->init_hooks();
```

Comment:
This feels a little risky since there's a chance the `wc_payments_http` filter could be used somewhere and that instance won't have `init_hooks` called on it. It'll only be a problem if the class being used instead extends `WC_Payments_Http` and is expecting the hooks to be registered in the constructor, but that sounds possible?

Reply:
Yep makes sense. I found an example of the filter being used at the returned class just implements the interface - which feels like the correct way to do this.